checkov/terraform/checks/graph_checks/azure/StorageContainerActivityLogsNotPublic.yaml

```diff
@@ -3,38 +3,69 @@ metadata:
   id: "CKV2_AZURE_8"
   category: "LOGGING"
 definition:
-  and:
-    - cond_type: filter
-      attribute: resource_type
-      value:
-        - azurerm_storage_account
-      operator: within
-    - cond_type: connection
-      resource_types:
-        - azurerm_monitor_activity_log_alert
-      connected_resource_types:
-        - azurerm_storage_account
-      operator: exists
-    - cond_type: attribute
-      resource_types:
-        - azurerm_monitor_activity_log_alert
-      attribute: criteria.resource_id
-      operator: exists
-    - or:
-        - cond_type: attribute
+  or:
+    - and:
+      - cond_type: filter
+        attribute: resource_type
+        value:
+          - azurerm_storage_container
+        operator: within
+
+      - cond_type: connection
+        resource_types:
+          - azurerm_storage_container
+        connected_resource_types:
+          - azurerm_storage_account
+        operator: exists
+
+      - or:
+        - cond_type: connection
           resource_types:
+            - azurerm_storage_account
+          connected_resource_types:
             - azurerm_monitor_activity_log_alert
-          attribute: enabled
           operator: not_exists
+
+        - and:
+          - cond_type: connection
+            resource_types:
+              - azurerm_storage_account
+            connected_resource_types:
+              - azurerm_monitor_activity_log_alert
+            operator: exists
+
+          - cond_type: attribute
+            resource_types:
+              - azurerm_monitor_activity_log_alert
+            attribute: enabled
+            operator: equals
+            value: false
+
+    - and:
+      - cond_type: filter
+        attribute: resource_type
+        value:
+          - azurerm_storage_container
+        operator: within
+
+      - cond_type: connection
+        resource_types:
+          - azurerm_storage_account
+        connected_resource_types:
+          - azurerm_storage_container
+          - azurerm_monitor_activity_log_alert
+        operator: exists
+
+      - or:
```

Comment:
I think the way you chose is easier to understand, but you could actually move this or block to the previous one at the end

Reply:
You are correct as always 👍 
fixed it.